test/e2e/e2e_test.go

```diff
@@ -1106,38 +1062,21 @@ var _ = Describe("e2e non-vxlan external gateway and update validation", func()
 		}
 		// Create the pod that will be used as the source for the connectivity test
 		createGenericPod(f, srcPingPodName, ciWorkerNodeSrc, command)
-
-		// Wait for pod exgw setup to be almost ready
-		wait.PollImmediate(1*time.Second, 30*time.Second, func() (bool, error) {
-			jsonFlag := "jsonpath='{.metadata.annotations.k8s\\.ovn\\.org/routing-external-gws}'"
-			kubectlOut, err := framework.RunKubectl("get", "pod", ovnNsFlag, srcPingPodName, jsonFlag)
-			if err != nil {
+		// wait for pod setup to return a valid address
+		err = wait.PollImmediate(retryInterval, retryTimeout, func() (bool, error) {
+			pingSrc, err = getPodAddress(srcPingPodName, f.Namespace.Name)
```

Comment:
same as comment above, check the err and return false to retry, I think that this will remove the parse ip check but it logs the error and the cause

Reply:
Skip the ParseIP validation because it is done upstream?

Reply:
nah, I was thinking that you can avoid parsing the IP, but you still have to check if it returned something.
It seems beter this way

Reply:
just only check for the error

Reply:
Added in the log statement, good call ty!